patroni/ha.py

```diff
@@ -252,12 +278,18 @@ def recover(self):
     def _get_node_to_follow(self, cluster):
         # determine the node to follow. If replicatefrom tag is set,
         # try to follow the node mentioned there, otherwise, follow the leader.
-        if not self.patroni.replicatefrom or self.patroni.replicatefrom == self.state_handler.name:
-            node_to_follow = cluster.leader
-        else:
+        is_leader = self.cluster.leader and self.state_handler.name == self.cluster.leader.name
+
+        if self.patroni.replicatefrom and self.patroni.replicatefrom != self.state_handler.name:
```

Comment:
What if node is the leader of standby cluster and replicatefrom tag is defined?
We will get a replication loop :)